utils.go

```diff
@@ -87,7 +87,13 @@ func SelectHopHints(ctx context.Context, lnd *lndclient.LndServices, amtMSat btc
 			continue
 		}
 
-		// TODO: HARSHA cache this
+		// If include_nodes is set, we'll only add channels with peers in include_node.
+		// This is done to respect the last_hop parameter
+		if _, ok := include_nodes[channel.PubKeyBytes]; include_nodes != nil && !ok {
```

Comment:
Since we're either going to have to keep this up to date with lnd or update to point to lnd, I think it would make sense to do our custom processing outside of this function.